

<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <title>Differential gene expression analysis &mdash; Inclass Activities 170218 documentation</title>
  

  
  
  
  

  

  
  
    

  

  
  
    <link rel="stylesheet" href="_static/css/theme.css" type="text/css" />
  

  

  
        <link rel="index" title="Index"
              href="genindex.html"/>
        <link rel="search" title="Search" href="search.html"/>
    <link rel="top" title="Inclass Activities 170218 documentation" href="index.html"/>
        <link rel="next" title="Secondary Structure Prediction" href="protein_structure.html"/>
        <link rel="prev" title="Counting mapped reads" href="flag_stat.html"/> 

  
  <script src="_static/js/modernizr.min.js"></script>

</head>

<body class="wy-body-for-nav" role="document">

   
  <div class="wy-grid-for-nav">

    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search">
          

          
            <a href="index.html" class="icon icon-home"> Inclass Activities
          

          
          </a>

          
            
            
              <div class="version">
                v1
              </div>
            
          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
              
            
            
              <p class="caption"><span class="caption-text">Bioinformatics part I</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="MEGA_alignment.html">1. Multiple Sequence Alignments</a></li>
<li class="toctree-l1"><a class="reference internal" href="phylogenetics.html">2. Steps of building a tree</a></li>
<li class="toctree-l1"><a class="reference internal" href="phylogenetics_2.html">3. Steps of building a tree (Part II)</a></li>
</ul>
<p class="caption"><span class="caption-text">High-throughput data analysis</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="fastQC.html">FastQC analysis using Cyverse Discovery Environment (DE)</a></li>
<li class="toctree-l1"><a class="reference internal" href="relaunch.html">Relaunching a stalled analysis in Cyverse Discovery Environment</a></li>
<li class="toctree-l1"><a class="reference internal" href="adapter_trimming.html">Adapter and quality trimming using trim-galore</a></li>
<li class="toctree-l1"><a class="reference internal" href="mapping.html">Mapping short reads</a></li>
<li class="toctree-l1"><a class="reference internal" href="flag_stat.html">Counting mapped reads</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">Differential gene expression analysis</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#deseq-tutorial">DESeq tutorial:</a></li>
<li class="toctree-l2"><a class="reference internal" href="#steps-to-perform-deseq-analysis">Steps to perform DEseq analysis</a></li>
<li class="toctree-l2"><a class="reference internal" href="#de-gene-list">DE gene list</a></li>
</ul>
</li>
</ul>
<p class="caption"><span class="caption-text">Protein analysis</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="protein_structure.html">Secondary Structure Prediction</a></li>
<li class="toctree-l1"><a class="reference internal" href="protein_structure.html#tertiary-structure-prediction">Tertiary Structure Prediction</a></li>
</ul>
<p class="caption"><span class="caption-text">qPCR data analysis</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="Delta-Delta_Ct_Method.html">The Delta-Delta Ct Method</a></li>
</ul>
<p class="caption"><span class="caption-text">Galaxy for HTS analysis</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="getting_data.html">Getting data into Galaxy</a></li>
<li class="toctree-l1"><a class="reference internal" href="GalaxyfastQC.html">FastQC analysis using Galaxy</a></li>
<li class="toctree-l1"><a class="reference internal" href="galaxy_adapter_trimming.html">Adapter and quality trimming using Cutadapt</a></li>
<li class="toctree-l1"><a class="reference internal" href="tophat.html">Use Splice aware aligner, Tophat2 to align short reads</a></li>
<li class="toctree-l1"><a class="reference internal" href="htseq.html">Use Htseq to counts reads mapped to features</a></li>
<li class="toctree-l1"><a class="reference internal" href="kellisto.html">Use Kellisto to map reads to cDNA and count</a></li>
<li class="toctree-l1"><a class="reference internal" href="DE_analysisR.html">Setup instructions (This is from Data Carpentry (http://www.datacarpentry.org/R-genomics/))</a></li>
<li class="toctree-l1"><a class="reference internal" href="DE_analysisR.html#using-deseq-and-edger-to-find-differentially-expressed-genes">Using DEseq and EdgeR to find differentially expressed genes</a></li>
<li class="toctree-l1"><a class="reference internal" href="DE_analysisR.html#deseq-analysis">DEseq analysis</a></li>
<li class="toctree-l1"><a class="reference internal" href="DE_analysisR.html#combine-deseq-and-edgr-to-make-venn-diagram">Combine DESeq and EdgR  to make Venn diagram</a></li>
<li class="toctree-l1"><a class="reference internal" href="GO_analysis.html">GOseq analysis</a></li>
</ul>

            
          
        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" role="navigation" aria-label="top navigation">
        
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="index.html">Inclass Activities</a>
        
      </nav>


      
      <div class="wy-nav-content">
        <div class="rst-content">
          















<div role="navigation" aria-label="breadcrumbs navigation">

  <ul class="wy-breadcrumbs">
    
      <li><a href="index.html">Docs</a> &raquo;</li>
        
      <li>Differential gene expression analysis</li>
    
    
      <li class="wy-breadcrumbs-aside">
        
            
            <a href="_sources/DE_analysis.rst.txt" rel="nofollow"> View page source</a>
          
        
      </li>
    
  </ul>

  
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  <div class="figure align-right">
<img alt="_images/UnivLogo_Stack_2C_Dark.png" src="_images/UnivLogo_Stack_2C_Dark.png" />
</div>
<div class="section" id="differential-gene-expression-analysis">
<h1>Differential gene expression analysis<a class="headerlink" href="#differential-gene-expression-analysis" title="Permalink to this headline">¶</a></h1>
<p>Link for  Bowtie mapped counts <a class="reference external" href="http://de.cyverse.org/dl/d/E9B4C299-D6CB-4656-A4F6-FF67240AEA49/170407_bowtie_counts.txt">http://de.cyverse.org/dl/d/E9B4C299-D6CB-4656-A4F6-FF67240AEA49/170407_bowtie_counts.txt</a></p>
<p>Target file for bowtie mapped reads: <a class="reference external" href="http://de.cyverse.org/dl/d/BECB62C3-A369-4084-9BC9-2BFD9E6E9600/bowtie_target.txt">http://de.cyverse.org/dl/d/BECB62C3-A369-4084-9BC9-2BFD9E6E9600/bowtie_target.txt</a></p>
<div class="section" id="deseq-tutorial">
<h2>DESeq tutorial:<a class="headerlink" href="#deseq-tutorial" title="Permalink to this headline">¶</a></h2>
<p><a class="reference external" href="https://pods.iplantcollaborative.org/wiki/display/DEapps/DESeq">Tutorial link</a></p>
</div>
<div class="section" id="steps-to-perform-deseq-analysis">
<h2>Steps to perform DEseq analysis<a class="headerlink" href="#steps-to-perform-deseq-analysis" title="Permalink to this headline">¶</a></h2>
<ol class="arabic">
<li><p class="first">From Apps select &#8220;DEseq (Multifactorial Comparison)</p>
<blockquote>
<div><div class="figure">
<img alt="_images/DEseq_1.png" src="_images/DEseq_1.png" />
</div>
</div></blockquote>
</li>
</ol>
<div class="line-block">
<div class="line"><br /></div>
</div>
<ol class="arabic" start="2">
<li><p class="first">Name your analysis and select a folder where your results need to be saved.</p>
<blockquote>
<div><div class="figure">
<img alt="_images/DEseq_2.png" src="_images/DEseq_2.png" />
</div>
</div></blockquote>
</li>
</ol>
<div class="line-block">
<div class="line"><br /></div>
</div>
<ol class="arabic" start="3">
<li><p class="first">Select the correct target file and the count file.</p>
<blockquote>
<div><div class="figure">
<img alt="_images/DEseq_3.png" src="_images/DEseq_3.png" />
</div>
</div></blockquote>
</li>
</ol>
<div class="line-block">
<div class="line"><br /></div>
</div>
<p>4. Give a name to the project. Reference biological condition should be &#8220;control&#8221; samples. Variable of
interest is &#8220;group&#8221; (Column header of the third column of the target file).</p>
<blockquote>
<div><div class="figure">
<img alt="_images/DEseq_4.png" src="_images/DEseq_4.png" />
</div>
</div></blockquote>
<div class="line-block">
<div class="line"><br /></div>
</div>
<ol class="arabic" start="5">
<li><p class="first">Set the significant threshold to 0.05 and launch the analysis.</p>
<blockquote>
<div><div class="figure">
<img alt="_images/DEseq_5.png" src="_images/DEseq_5.png" />
</div>
</div></blockquote>
</li>
</ol>
</div>
<div class="section" id="de-gene-list">
<h2>DE gene list<a class="headerlink" href="#de-gene-list" title="Permalink to this headline">¶</a></h2>
<p>I have used the following R code to merge the DE genes list and the functions.</p>
<div class="highlight-R"><div class="highlight"><pre><span></span><span class="kn">library</span><span class="p">(</span>reshape2<span class="p">)</span>
<span class="kn">library</span><span class="p">(</span>readr<span class="p">)</span>

<span class="c1"># Used the terminal command to grep the fasta headers and wrote it to a file called &quot;ITAG3_10_cDN_names.txt&quot;</span>
<span class="c1">#imported this file to Rstudio</span>

<span class="c1"># Removed the &quot;&gt;&quot; sign</span>
ITAG3_10_formated_names <span class="o">&lt;-</span> <span class="kp">as.data.frame</span><span class="p">(</span><span class="kp">sapply</span><span class="p">(</span>ITAG3_10_cDN_names<span class="p">,</span> <span class="kp">gsub</span><span class="p">,</span> pattern <span class="o">=</span> <span class="s">&quot;&gt;&quot;</span><span class="p">,</span> replacement <span class="o">=</span> <span class="s">&quot;&quot;</span> <span class="p">))</span>

<span class="c1">#Seperate gene ids and description using space as delimiter</span>
ITAG3_10_formated_names <span class="o">&lt;-</span> <span class="kt">data.frame</span><span class="p">(</span>colsplit<span class="p">(</span>ITAG3_10_formated_names<span class="o">$</span>X1<span class="p">,</span> <span class="s">&quot; &quot;</span><span class="p">,</span> <span class="kt">c</span><span class="p">(</span><span class="s">&quot;Id&quot;</span><span class="p">,</span> <span class="s">&quot;Description&quot;</span><span class="p">)))</span>

<span class="c1">#imported up regulated genes to Rstudio and merge with the above file using gene ids.</span>
heatvscontrol_up_func <span class="o">&lt;-</span> <span class="kp">merge</span><span class="p">(</span>heatvscontrol_up<span class="p">,</span> ITAG3_10_formated_names<span class="p">,</span>
                       by.x <span class="o">=</span> <span class="s">&quot;Id&quot;</span><span class="p">,</span>
                       by.y <span class="o">=</span> <span class="s">&quot;Id&quot;</span><span class="p">)</span>

<span class="c1">#write output</span>
write.table<span class="p">(</span>x <span class="o">=</span> heatvscontrol_up_func<span class="p">,</span> file <span class="o">=</span> <span class="s">&quot;heatvscontrol_up_func.txt&quot;</span><span class="p">,</span> quote <span class="o">=</span> <span class="kc">FALSE</span><span class="p">,</span> sep <span class="o">=</span> <span class="s">&quot;\t&quot;</span><span class="p">,</span> row.names <span class="o">=</span> <span class="kc">FALSE</span><span class="p">)</span>


<span class="c1">#imported down regulated genes to Rstudio and merge with the above file using gene ids</span>
heatvscontrol_down_func <span class="o">&lt;-</span> <span class="kp">merge</span><span class="p">(</span>heatvscontrol_down<span class="p">,</span> ITAG3_10_formated_names<span class="p">,</span>
                       by.x <span class="o">=</span> <span class="s">&quot;Id&quot;</span><span class="p">,</span>
                       by.y <span class="o">=</span> <span class="s">&quot;Id&quot;</span><span class="p">)</span>
<span class="c1">#write output</span>
write.table<span class="p">(</span>x <span class="o">=</span> heatvscontrol_down_func<span class="p">,</span> file <span class="o">=</span> <span class="s">&quot;heatvscontrol_down_func.txt&quot;</span><span class="p">,</span> quote <span class="o">=</span> <span class="kc">FALSE</span><span class="p">,</span> sep <span class="o">=</span> <span class="s">&quot;\t&quot;</span><span class="p">,</span> row.names <span class="o">=</span> <span class="kc">FALSE</span><span class="p">)</span>
</pre></div>
</div>
<p>Up-regulated gene list: <a class="reference external" href="http://de.cyverse.org/dl/d/E641698E-8688-4C20-B829-0B12BABC8ABB/heatvscontrol_up_func.txt">http://de.cyverse.org/dl/d/E641698E-8688-4C20-B829-0B12BABC8ABB/heatvscontrol_up_func.txt</a></p>
<p>Down-regulated gene list: <a class="reference external" href="http://de.cyverse.org/dl/d/3C45B913-612F-4B97-8F44-8021470AE527/heatvscontrol_down_func.txt">http://de.cyverse.org/dl/d/3C45B913-612F-4B97-8F44-8021470AE527/heatvscontrol_down_func.txt</a></p>
</div>
</div>


           </div>
           <div class="articleComments">
            
           </div>
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
        <a href="protein_structure.html" class="btn btn-neutral float-right" title="Secondary Structure Prediction" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right"></span></a>
      
      
        <a href="flag_stat.html" class="btn btn-neutral" title="Counting mapped reads" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <p>
        &copy; Copyright 2017, Asela Wijeratne.

    </p>
  </div>
  Built with <a href="http://sphinx-doc.org/">Sphinx</a> using a <a href="https://github.com/snide/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>. 

</footer>

        </div>
      </div>

    </section>

  </div>
  


  

    <script type="text/javascript">
        var DOCUMENTATION_OPTIONS = {
            URL_ROOT:'./',
            VERSION:'170218',
            COLLAPSE_INDEX:false,
            FILE_SUFFIX:'.html',
            HAS_SOURCE:  true,
            SOURCELINK_SUFFIX: '.txt'
        };
    </script>
      <script type="text/javascript" src="_static/jquery.js"></script>
      <script type="text/javascript" src="_static/underscore.js"></script>
      <script type="text/javascript" src="_static/doctools.js"></script>

  

  
  
    <script type="text/javascript" src="_static/js/theme.js"></script>
  

  
  
  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.StickyNav.enable();
      });
  </script>
   

</body>
</html>