

<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <title>The Delta-Delta Ct Method &mdash; Inclass Activities 170218 documentation</title>
  

  
  
  
  

  

  
  
    

  

  
  
    <link rel="stylesheet" href="_static/css/theme.css" type="text/css" />
  

  

  
        <link rel="index" title="Index"
              href="genindex.html"/>
        <link rel="search" title="Search" href="search.html"/>
    <link rel="top" title="Inclass Activities 170218 documentation" href="index.html"/>
        <link rel="next" title="Getting data into Galaxy" href="getting_data.html"/>
        <link rel="prev" title="Secondary Structure Prediction" href="protein_structure.html"/> 

  
  <script src="_static/js/modernizr.min.js"></script>

</head>

<body class="wy-body-for-nav" role="document">

   
  <div class="wy-grid-for-nav">

    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search">
          

          
            <a href="index.html" class="icon icon-home"> Inclass Activities
          

          
          </a>

          
            
            
              <div class="version">
                v1
              </div>
            
          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
              
            
            
              <p class="caption"><span class="caption-text">Bioinformatics part I</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="MEGA_alignment.html">1. Multiple Sequence Alignments</a></li>
<li class="toctree-l1"><a class="reference internal" href="phylogenetics.html">2. Steps of building a tree</a></li>
<li class="toctree-l1"><a class="reference internal" href="phylogenetics_2.html">3. Steps of building a tree (Part II)</a></li>
</ul>
<p class="caption"><span class="caption-text">High-throughput data analysis</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="fastQC.html">FastQC analysis using Cyverse Discovery Environment (DE)</a></li>
<li class="toctree-l1"><a class="reference internal" href="relaunch.html">Relaunching a stalled analysis in Cyverse Discovery Environment</a></li>
<li class="toctree-l1"><a class="reference internal" href="adapter_trimming.html">Adapter and quality trimming using trim-galore</a></li>
<li class="toctree-l1"><a class="reference internal" href="mapping.html">Mapping short reads</a></li>
<li class="toctree-l1"><a class="reference internal" href="flag_stat.html">Counting mapped reads</a></li>
<li class="toctree-l1"><a class="reference internal" href="DE_analysis.html">Differential gene expression analysis</a></li>
</ul>
<p class="caption"><span class="caption-text">Protein analysis</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="protein_structure.html">Secondary Structure Prediction</a></li>
<li class="toctree-l1"><a class="reference internal" href="protein_structure.html#tertiary-structure-prediction">Tertiary Structure Prediction</a></li>
</ul>
<p class="caption"><span class="caption-text">qPCR data analysis</span></p>
<ul class="current">
<li class="toctree-l1 current"><a class="current reference internal" href="#">The Delta-Delta Ct Method</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#normalization">Normalization</a></li>
<li class="toctree-l2"><a class="reference internal" href="#average-of-the-control-samples-normal-cells">Average of the control samples (normal cells)</a></li>
<li class="toctree-l2"><a class="reference internal" href="#calculate-the-ct-relative-to-the-average-of-ct-normal-cells">Calculate the ∆∆Ct relative to the average of ∆Ct normal cells</a></li>
<li class="toctree-l2"><a class="reference internal" href="#fold-gene-expression-for-each-sample">Fold gene expression for each sample</a></li>
<li class="toctree-l2"><a class="reference internal" href="#overall-fold-change">Overall fold change</a></li>
<li class="toctree-l2"><a class="reference internal" href="#log-transformation">Log transformation</a></li>
<li class="toctree-l2"><a class="reference internal" href="#t-test">T-test</a></li>
</ul>
</li>
</ul>
<p class="caption"><span class="caption-text">Galaxy for HTS analysis</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="getting_data.html">Getting data into Galaxy</a></li>
<li class="toctree-l1"><a class="reference internal" href="GalaxyfastQC.html">FastQC analysis using Galaxy</a></li>
<li class="toctree-l1"><a class="reference internal" href="galaxy_adapter_trimming.html">Adapter and quality trimming using Cutadapt</a></li>
<li class="toctree-l1"><a class="reference internal" href="tophat.html">Use Splice aware aligner, Tophat2 to align short reads</a></li>
<li class="toctree-l1"><a class="reference internal" href="htseq.html">Use Htseq to counts reads mapped to features</a></li>
<li class="toctree-l1"><a class="reference internal" href="kellisto.html">Use Kellisto to map reads to cDNA and count</a></li>
<li class="toctree-l1"><a class="reference internal" href="DE_analysisR.html">Setup instructions (This is from Data Carpentry (http://www.datacarpentry.org/R-genomics/))</a></li>
<li class="toctree-l1"><a class="reference internal" href="DE_analysisR.html#using-deseq-and-edger-to-find-differentially-expressed-genes">Using DEseq and EdgeR to find differentially expressed genes</a></li>
<li class="toctree-l1"><a class="reference internal" href="DE_analysisR.html#deseq-analysis">DEseq analysis</a></li>
<li class="toctree-l1"><a class="reference internal" href="DE_analysisR.html#combine-deseq-and-edgr-to-make-venn-diagram">Combine DESeq and EdgR  to make Venn diagram</a></li>
<li class="toctree-l1"><a class="reference internal" href="GO_analysis.html">GOseq analysis</a></li>
<li class="toctree-l1"><a class="reference internal" href="workflow.html">Run RNAseq analysis as a workflow</a></li>
</ul>

            
          
        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" role="navigation" aria-label="top navigation">
        
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="index.html">Inclass Activities</a>
        
      </nav>


      
      <div class="wy-nav-content">
        <div class="rst-content">
          















<div role="navigation" aria-label="breadcrumbs navigation">

  <ul class="wy-breadcrumbs">
    
      <li><a href="index.html">Docs</a> &raquo;</li>
        
      <li>The Delta-Delta Ct Method</li>
    
    
      <li class="wy-breadcrumbs-aside">
        
            
            <a href="_sources/Delta-Delta_Ct_Method.rst.txt" rel="nofollow"> View page source</a>
          
        
      </li>
    
  </ul>

  
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  <div class="figure align-right">
<img alt="_images/UnivLogo_Stack_2C_Dark.png" src="_images/UnivLogo_Stack_2C_Dark.png" />
</div>
<div class="section" id="the-delta-delta-ct-method">
<h1>The Delta-Delta Ct Method<a class="headerlink" href="#the-delta-delta-ct-method" title="Permalink to this headline">¶</a></h1>
<p>Delta-Delta Ct method or Livak method is the most preferred method for qPCR data analysis. However, it can only be used when certain criteria are met.
Please refer the lecture notes to make sure that these criteria are fulfilled. If not, more generalized method is called Pfaffl method. Please read the additional reading material to
get more information about this method.</p>
<p>Here are the steps for Livak method:</p>
<p>The Excel file with all the calculation are in the qPCR analysis folder on Blackboard.</p>
<p>You have raw Ct (number of cycles that takes to reach threshold) for normal and tumor cells (3 replicates for each).</p>
<blockquote>
<div><div class="figure">
<img alt="_images/qPCR_1.png" src="_images/qPCR_1.png" />
</div>
</div></blockquote>
<hr class="docutils" />
<div class="section" id="normalization">
<h2>Normalization<a class="headerlink" href="#normalization" title="Permalink to this headline">¶</a></h2>
<p>First, you will need calculate relative difference between the gene of interest (p53) and the house keeping gene (GAPDH).</p>
<p>∆Ct = Ct (gene of interest) – Ct (housekeeping gene)</p>
<blockquote>
<div><div class="figure">
<img alt="_images/qPCR_2.png" src="_images/qPCR_2.png" />
</div>
</div></blockquote>
</div>
<hr class="docutils" />
<div class="section" id="average-of-the-control-samples-normal-cells">
<h2>Average of the control samples (normal cells)<a class="headerlink" href="#average-of-the-control-samples-normal-cells" title="Permalink to this headline">¶</a></h2>
<p>As we compare our tumor (treatment) to control (normal cells), first we need to average the ∆Ct for the 3 control (normal) samples.</p>
<blockquote>
<div><div class="figure">
<img alt="_images/qPCR_3.png" src="_images/qPCR_3.png" />
</div>
</div></blockquote>
</div>
<hr class="docutils" />
<div class="section" id="calculate-the-ct-relative-to-the-average-of-ct-normal-cells">
<h2>Calculate the ∆∆Ct relative to the average of ∆Ct normal cells<a class="headerlink" href="#calculate-the-ct-relative-to-the-average-of-ct-normal-cells" title="Permalink to this headline">¶</a></h2>
<p>∆∆Ct = ∆Ct (Tumor sample) – ∆Ct (normal average)</p>
<p>You can do this normal samples as well. Use $ signs infront of column number and raw letter (arrows) to fix the cell.</p>
<blockquote>
<div><div class="figure">
<img alt="_images/qPCR_4.png" src="_images/qPCR_4.png" />
</div>
</div></blockquote>
</div>
<hr class="docutils" />
<div class="section" id="fold-gene-expression-for-each-sample">
<h2>Fold gene expression for each sample<a class="headerlink" href="#fold-gene-expression-for-each-sample" title="Permalink to this headline">¶</a></h2>
<p>Make sure you raise the negative ∆∆Ct to power of two.</p>
<p>Fold gene expression = 2^-(∆∆Ct)</p>
<blockquote>
<div><div class="figure">
<img alt="_images/qPCR_5.png" src="_images/qPCR_5.png" />
</div>
</div></blockquote>
</div>
<hr class="docutils" />
<div class="section" id="overall-fold-change">
<h2>Overall fold change<a class="headerlink" href="#overall-fold-change" title="Permalink to this headline">¶</a></h2>
<p>You can calculate average fold change for both tumor and normal samples. Ratio between these two the fold change between tumor and normal samples.</p>
<blockquote>
<div><div class="figure">
<img alt="_images/qPCR_6.png" src="_images/qPCR_6.png" />
</div>
<div class="figure">
<img alt="_images/qPCR_7.png" src="_images/qPCR_7.png" />
</div>
<div class="figure">
<img alt="_images/qPCR_8.png" src="_images/qPCR_8.png" />
</div>
</div></blockquote>
</div>
<hr class="docutils" />
<div class="section" id="log-transformation">
<h2>Log transformation<a class="headerlink" href="#log-transformation" title="Permalink to this headline">¶</a></h2>
<p>To perform  parametric statistical tests such as T-test, it advised to transform the final gene expression results to log values (any log base). This would make
data distribution symmetric.</p>
<p>Here we have change the 2^-(∆∆Ct) to log 10.</p>
<blockquote>
<div><div class="figure">
<img alt="_images/qPCR_9.png" src="_images/qPCR_9.png" />
</div>
</div></blockquote>
</div>
<hr class="docutils" />
<div class="section" id="t-test">
<h2>T-test<a class="headerlink" href="#t-test" title="Permalink to this headline">¶</a></h2>
<p>Need to be careful when using parametric tests if data is not normally distributed, it would lead to erroneous conclusions.</p>
<blockquote>
<div><div class="figure">
<img alt="_images/qPCR_9.png" src="_images/qPCR_9.png" />
</div>
</div></blockquote>
<p>Select log 10 of 2^-(∆∆Ct) values for Normal and tumor samples as indicated. Use two tail test (number 2) and assuming unequal variance (3).</p>
<p>Resulting P value is less than 0.05 and therefore, we reject the null hypothesis and two sample means are significantly different at 0.05 level.</p>
<blockquote>
<div><div class="figure">
<img alt="_images/qPCR_11.png" src="_images/qPCR_11.png" />
</div>
</div></blockquote>
</div>
</div>


           </div>
           <div class="articleComments">
            
           </div>
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
        <a href="getting_data.html" class="btn btn-neutral float-right" title="Getting data into Galaxy" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right"></span></a>
      
      
        <a href="protein_structure.html" class="btn btn-neutral" title="Secondary Structure Prediction" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <p>
        &copy; Copyright 2017, Asela Wijeratne.

    </p>
  </div>
  Built with <a href="http://sphinx-doc.org/">Sphinx</a> using a <a href="https://github.com/snide/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>. 

</footer>

        </div>
      </div>

    </section>

  </div>
  


  

    <script type="text/javascript">
        var DOCUMENTATION_OPTIONS = {
            URL_ROOT:'./',
            VERSION:'170218',
            COLLAPSE_INDEX:false,
            FILE_SUFFIX:'.html',
            HAS_SOURCE:  true,
            SOURCELINK_SUFFIX: '.txt'
        };
    </script>
      <script type="text/javascript" src="_static/jquery.js"></script>
      <script type="text/javascript" src="_static/underscore.js"></script>
      <script type="text/javascript" src="_static/doctools.js"></script>

  

  
  
    <script type="text/javascript" src="_static/js/theme.js"></script>
  

  
  
  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.StickyNav.enable();
      });
  </script>
   

</body>
</html>