

<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <title>Counting mapped reads &mdash; Inclass Activities 170218 documentation</title>
  

  
  
  
  

  

  
  
    

  

  
  
    <link rel="stylesheet" href="_static/css/theme.css" type="text/css" />
  

  

  
        <link rel="index" title="Index"
              href="genindex.html"/>
        <link rel="search" title="Search" href="search.html"/>
    <link rel="top" title="Inclass Activities 170218 documentation" href="index.html"/>
        <link rel="next" title="Differential gene expression analysis" href="DE_analysis.html"/>
        <link rel="prev" title="Mapping short reads" href="mapping.html"/> 

  
  <script src="_static/js/modernizr.min.js"></script>

</head>

<body class="wy-body-for-nav" role="document">

   
  <div class="wy-grid-for-nav">

    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search">
          

          
            <a href="index.html" class="icon icon-home"> Inclass Activities
          

          
          </a>

          
            
            
              <div class="version">
                v1
              </div>
            
          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
              
            
            
              <p class="caption"><span class="caption-text">Bioinformatics part I</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="MEGA_alignment.html">1. Multiple Sequence Alignments</a></li>
<li class="toctree-l1"><a class="reference internal" href="phylogenetics.html">2. Steps of building a tree</a></li>
<li class="toctree-l1"><a class="reference internal" href="phylogenetics_2.html">3. Steps of building a tree (Part II)</a></li>
</ul>
<p class="caption"><span class="caption-text">High-throughput data analysis</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="fastQC.html">FastQC analysis using Cyverse Discovery Environment (DE)</a></li>
<li class="toctree-l1"><a class="reference internal" href="relaunch.html">Relaunching a stalled analysis in Cyverse Discovery Environment</a></li>
<li class="toctree-l1"><a class="reference internal" href="adapter_trimming.html">Adapter and quality trimming using trim-galore</a></li>
<li class="toctree-l1"><a class="reference internal" href="mapping.html">Mapping short reads</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">Counting mapped reads</a></li>
<li class="toctree-l1"><a class="reference internal" href="DE_analysis.html">Differential gene expression analysis</a></li>
</ul>
<p class="caption"><span class="caption-text">Protein analysis</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="protein_structure.html">Secondary Structure Prediction</a></li>
<li class="toctree-l1"><a class="reference internal" href="protein_structure.html#tertiary-structure-prediction">Tertiary Structure Prediction</a></li>
</ul>
<p class="caption"><span class="caption-text">qPCR data analysis</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="Delta-Delta_Ct_Method.html">The Delta-Delta Ct Method</a></li>
</ul>
<p class="caption"><span class="caption-text">Galaxy for HTS analysis</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="getting_data.html">Getting data into Galaxy</a></li>
<li class="toctree-l1"><a class="reference internal" href="GalaxyfastQC.html">FastQC analysis using Galaxy</a></li>
<li class="toctree-l1"><a class="reference internal" href="galaxy_adapter_trimming.html">Adapter and quality trimming using Cutadapt</a></li>
<li class="toctree-l1"><a class="reference internal" href="tophat.html">Use Splice aware aligner, Tophat2 to align short reads</a></li>
<li class="toctree-l1"><a class="reference internal" href="htseq.html">Use Htseq to counts reads mapped to features</a></li>
<li class="toctree-l1"><a class="reference internal" href="kellisto.html">Use Kellisto to map reads to cDNA and count</a></li>
<li class="toctree-l1"><a class="reference internal" href="DE_analysisR.html">Setup instructions (This is from Data Carpentry (http://www.datacarpentry.org/R-genomics/))</a></li>
<li class="toctree-l1"><a class="reference internal" href="DE_analysisR.html#using-deseq-and-edger-to-find-differentially-expressed-genes">Using DEseq and EdgeR to find differentially expressed genes</a></li>
<li class="toctree-l1"><a class="reference internal" href="DE_analysisR.html#deseq-analysis">DEseq analysis</a></li>
<li class="toctree-l1"><a class="reference internal" href="DE_analysisR.html#combine-deseq-and-edgr-to-make-venn-diagram">Combine DESeq and EdgR  to make Venn diagram</a></li>
<li class="toctree-l1"><a class="reference internal" href="GO_analysis.html">GOseq analysis</a></li>
<li class="toctree-l1"><a class="reference internal" href="workflow.html">Run RNAseq analysis as a workflow</a></li>
</ul>

            
          
        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" role="navigation" aria-label="top navigation">
        
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="index.html">Inclass Activities</a>
        
      </nav>


      
      <div class="wy-nav-content">
        <div class="rst-content">
          















<div role="navigation" aria-label="breadcrumbs navigation">

  <ul class="wy-breadcrumbs">
    
      <li><a href="index.html">Docs</a> &raquo;</li>
        
      <li>Counting mapped reads</li>
    
    
      <li class="wy-breadcrumbs-aside">
        
            
            <a href="_sources/flag_stat.rst.txt" rel="nofollow"> View page source</a>
          
        
      </li>
    
  </ul>

  
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  <div class="figure align-right">
<img alt="_images/UnivLogo_Stack_2C_Dark.png" src="_images/UnivLogo_Stack_2C_Dark.png" />
</div>
<div class="section" id="counting-mapped-reads">
<h1>Counting mapped reads<a class="headerlink" href="#counting-mapped-reads" title="Permalink to this headline">Â¶</a></h1>
<p>To get the number of reads mapped to a reference sequences (in this case, predicted tomato cDNA sequences), we can use Samtools. Bowtie2
output is in sam format and first, we need to convert the output files into sorted bam files.</p>
<ol class="arabic simple">
<li>Type Samtools in app finding window.</li>
</ol>
<div class="figure">
<img alt="_images/samtools_1.png" src="_images/samtools_1.png" />
</div>
<ol class="arabic simple" start="2">
<li>Select &#8220;SAM to sorted BAM&#8221;</li>
</ol>
<div class="figure">
<img alt="_images/samtools_2.png" src="_images/samtools_2.png" />
</div>
<ol class="arabic simple" start="3">
<li>Select Bowtie2 output files (SAM format).</li>
</ol>
<div class="figure">
<img alt="_images/samtools_3.png" src="_images/samtools_3.png" />
</div>
<p>4. Above will create sorted bam file. You will need to use this as the input for the Samtools Flagstat, which
will count the number of mapped reads.</p>
<div class="figure">
<img alt="_images/samtools_4.png" src="_images/samtools_4.png" />
</div>
<div class="line-block">
<div class="line"><br /></div>
</div>
<div class="figure">
<img alt="_images/samtools_5.png" src="_images/samtools_5.png" />
</div>
<p>** You can get the flagstat for all six files from following link.**
|
<a class="reference external" href="https://github.com/ajwije/2017_spring_Bioinfo_class/blob/master/Files/flagstat.txt">https://github.com/ajwije/2017_spring_Bioinfo_class/blob/master/Files/flagstat.txt</a></p>
<p>I have used the following bash command to count mapped reads in case you are interested in it doing programmatically.</p>
<div class="highlight-bash"><div class="highlight"><pre><span></span><span class="c1"># navigate to where your sam files are and execute the following commands.</span>

<span class="k">for</span> i in *.sam

<span class="k">do</span>
<span class="c1">#extract the file name without extension and print it to the screen</span>


<span class="nb">echo</span> <span class="si">${</span><span class="nv">i</span><span class="p">%.sam</span><span class="si">}</span>

<span class="c1">#covert sam to sorted bam</span>
samtools view -bS <span class="nv">$i</span> <span class="p">|</span> samtools sort - -o <span class="si">${</span><span class="nv">i</span><span class="p">%.sam</span><span class="si">}</span>_sorted.bam

<span class="c1">#getting flagstat and write it an output file for each bam file.</span>
samtools flagstat <span class="si">${</span><span class="nv">fname</span><span class="p">%.sam</span><span class="si">}</span>_sorted.bam &gt;&gt; flagstat.txt

<span class="k">done</span>
</pre></div>
</div>
</div>


           </div>
           <div class="articleComments">
            
           </div>
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
        <a href="DE_analysis.html" class="btn btn-neutral float-right" title="Differential gene expression analysis" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right"></span></a>
      
      
        <a href="mapping.html" class="btn btn-neutral" title="Mapping short reads" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <p>
        &copy; Copyright 2017, Asela Wijeratne.

    </p>
  </div>
  Built with <a href="http://sphinx-doc.org/">Sphinx</a> using a <a href="https://github.com/snide/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>. 

</footer>

        </div>
      </div>

    </section>

  </div>
  


  

    <script type="text/javascript">
        var DOCUMENTATION_OPTIONS = {
            URL_ROOT:'./',
            VERSION:'170218',
            COLLAPSE_INDEX:false,
            FILE_SUFFIX:'.html',
            HAS_SOURCE:  true,
            SOURCELINK_SUFFIX: '.txt'
        };
    </script>
      <script type="text/javascript" src="_static/jquery.js"></script>
      <script type="text/javascript" src="_static/underscore.js"></script>
      <script type="text/javascript" src="_static/doctools.js"></script>

  

  
  
    <script type="text/javascript" src="_static/js/theme.js"></script>
  

  
  
  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.StickyNav.enable();
      });
  </script>
   

</body>
</html>