

<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <title>GOseq analysis &mdash; Inclass Activities 170218 documentation</title>
  

  
  
  
  

  

  
  
    

  

  
  
    <link rel="stylesheet" href="_static/css/theme.css" type="text/css" />
  

  

  
        <link rel="index" title="Index"
              href="genindex.html"/>
        <link rel="search" title="Search" href="search.html"/>
    <link rel="top" title="Inclass Activities 170218 documentation" href="index.html"/>
        <link rel="prev" title="Setup instructions (This is from Data Carpentry (http://www.datacarpentry.org/R-genomics/))" href="DE_analysisR.html"/> 

  
  <script src="_static/js/modernizr.min.js"></script>

</head>

<body class="wy-body-for-nav" role="document">

   
  <div class="wy-grid-for-nav">

    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search">
          

          
            <a href="index.html" class="icon icon-home"> Inclass Activities
          

          
          </a>

          
            
            
              <div class="version">
                v1
              </div>
            
          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
              
            
            
              <p class="caption"><span class="caption-text">Bioinformatics part I</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="MEGA_alignment.html">1. Multiple Sequence Alignments</a></li>
<li class="toctree-l1"><a class="reference internal" href="phylogenetics.html">2. Steps of building a tree</a></li>
<li class="toctree-l1"><a class="reference internal" href="phylogenetics_2.html">3. Steps of building a tree (Part II)</a></li>
</ul>
<p class="caption"><span class="caption-text">High-throughput data analysis</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="fastQC.html">FastQC analysis using Cyverse Discovery Environment (DE)</a></li>
<li class="toctree-l1"><a class="reference internal" href="relaunch.html">Relaunching a stalled analysis in Cyverse Discovery Environment</a></li>
<li class="toctree-l1"><a class="reference internal" href="adapter_trimming.html">Adapter and quality trimming using trim-galore</a></li>
<li class="toctree-l1"><a class="reference internal" href="mapping.html">Mapping short reads</a></li>
<li class="toctree-l1"><a class="reference internal" href="flag_stat.html">Counting mapped reads</a></li>
<li class="toctree-l1"><a class="reference internal" href="DE_analysis.html">Differential gene expression analysis</a></li>
</ul>
<p class="caption"><span class="caption-text">Protein analysis</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="protein_structure.html">Secondary Structure Prediction</a></li>
<li class="toctree-l1"><a class="reference internal" href="protein_structure.html#tertiary-structure-prediction">Tertiary Structure Prediction</a></li>
</ul>
<p class="caption"><span class="caption-text">qPCR data analysis</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="Delta-Delta_Ct_Method.html">The Delta-Delta Ct Method</a></li>
</ul>
<p class="caption"><span class="caption-text">Galaxy for HTS analysis</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="getting_data.html">Getting data into Galaxy</a></li>
<li class="toctree-l1"><a class="reference internal" href="GalaxyfastQC.html">FastQC analysis using Galaxy</a></li>
<li class="toctree-l1"><a class="reference internal" href="galaxy_adapter_trimming.html">Adapter and quality trimming using Cutadapt</a></li>
<li class="toctree-l1"><a class="reference internal" href="tophat.html">Use Splice aware aligner, Tophat2 to align short reads</a></li>
<li class="toctree-l1"><a class="reference internal" href="htseq.html">Use Htseq to counts reads mapped to features</a></li>
<li class="toctree-l1"><a class="reference internal" href="kellisto.html">Use Kellisto to map reads to cDNA and count</a></li>
<li class="toctree-l1"><a class="reference internal" href="DE_analysisR.html">Setup instructions (This is from Data Carpentry (http://www.datacarpentry.org/R-genomics/))</a></li>
<li class="toctree-l1"><a class="reference internal" href="DE_analysisR.html#using-deseq-and-edger-to-find-differentially-expressed-genes">Using DEseq and EdgeR to find differentially expressed genes</a></li>
<li class="toctree-l1"><a class="reference internal" href="DE_analysisR.html#deseq-analysis">DEseq analysis</a></li>
<li class="toctree-l1"><a class="reference internal" href="DE_analysisR.html#combine-deseq-and-edgr-to-make-venn-diagram">Combine DESeq and EdgR  to make Venn diagram</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">GOseq analysis</a></li>
</ul>

            
          
        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" role="navigation" aria-label="top navigation">
        
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="index.html">Inclass Activities</a>
        
      </nav>


      
      <div class="wy-nav-content">
        <div class="rst-content">
          















<div role="navigation" aria-label="breadcrumbs navigation">

  <ul class="wy-breadcrumbs">
    
      <li><a href="index.html">Docs</a> &raquo;</li>
        
      <li>GOseq analysis</li>
    
    
      <li class="wy-breadcrumbs-aside">
        
            
            <a href="_sources/GO_analysis.rst.txt" rel="nofollow"> View page source</a>
          
        
      </li>
    
  </ul>

  
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  <div class="figure align-right">
<img alt="_images/UnivLogo_Stack_2C_Dark.png" src="_images/UnivLogo_Stack_2C_Dark.png" />
</div>
<hr class="docutils" />
<div class="section" id="goseq-analysis">
<h1>GOseq analysis<a class="headerlink" href="#goseq-analysis" title="Permalink to this headline">Â¶</a></h1>
<p>1. Ge the up and down regulated gene list.
&#8220;bothDF&#8221; is the dataframe that contains both up and down-regulated genes from both EdgeR and DEseq2.</p>
<div class="highlight-r"><div class="highlight"><pre><span></span>bothDF_down <span class="o">&lt;-</span> bothDF<span class="p">[</span>bothDF<span class="o">$</span>log2FoldChange <span class="o">&lt;=</span> <span class="m">-1</span><span class="p">,]</span>
bothDF_up <span class="o">&lt;-</span> bothDF<span class="p">[</span>bothDF<span class="o">$</span>log2FoldChange <span class="o">&gt;=</span> <span class="m">1</span><span class="p">,]</span>
</pre></div>
</div>
<p>Convert to these dataframes into table with True or False values. Write the table to local directory.</p>
<div class="highlight-r"><div class="highlight"><pre><span></span>DE_list_boolup <span class="o">&lt;-</span> <span class="kp">as.data.frame</span><span class="p">(</span><span class="kp">row.names</span><span class="p">(</span>d<span class="p">)</span> <span class="o">%in%</span> bothDF_up<span class="o">$</span>Row.names<span class="p">,</span>
      row.names <span class="o">=</span> <span class="kp">row.names</span><span class="p">(</span>d<span class="p">))</span>
DE_list_booldown <span class="o">&lt;-</span> <span class="kp">as.data.frame</span><span class="p">(</span><span class="kp">row.names</span><span class="p">(</span>d<span class="p">)</span> <span class="o">%in%</span> bothDF_down<span class="o">$</span>Row.names<span class="p">,</span>
                        row.names <span class="o">=</span> <span class="kp">row.names</span><span class="p">(</span>d<span class="p">))</span>

write.table<span class="p">(</span>DE_list_boolup<span class="p">,</span>file<span class="o">=</span><span class="s">&quot;DE_goseq_up.txt&quot;</span><span class="p">,</span>row.names<span class="o">=</span><span class="bp">T</span><span class="p">,</span>sep<span class="o">=</span><span class="s">&#39;\t&#39;</span><span class="p">,</span>quote<span class="o">=</span><span class="bp">F</span><span class="p">,</span>
    col.names <span class="o">=</span> <span class="bp">F</span><span class="p">)</span>

write.table<span class="p">(</span>DE_list_booldown<span class="p">,</span>file<span class="o">=</span><span class="s">&quot;DE_goseq_down.txt&quot;</span><span class="p">,</span>row.names<span class="o">=</span><span class="bp">T</span><span class="p">,</span>sep<span class="o">=</span><span class="s">&#39;\t&#39;</span><span class="p">,</span>quote<span class="o">=</span><span class="bp">F</span><span class="p">,</span>
col.names <span class="o">=</span> <span class="bp">F</span><span class="p">)</span>
</pre></div>
</div>
<ol class="arabic" start="2">
<li><p class="first">Perform the GOseq analysis in Galaxy. You will need to perform the analysis for up and down regulated genes separately.</p>
<blockquote>
<div><div class="figure">
<img alt="_images/GOseq.png" src="_images/GOseq.png" />
</div>
</div></blockquote>
</li>
</ol>
<hr class="docutils" />
<ol class="arabic simple" start="3">
<li>Combine gene descriptions with up and down regulated genes. You can get the S_lycopersicum_Feb_2014.bed file from the Dropbox link on Bb.</li>
</ol>
<div class="highlight-r"><div class="highlight"><pre><span></span>annots_file <span class="o">&lt;-</span> <span class="s">&#39;S_lycopersicum_Feb_2014.bed&#39;</span>
<span class="c1"># keep gene id and gene description columns</span>
annots <span class="o">&lt;-</span> read.delim<span class="p">(</span>annots_file<span class="p">,</span>sep<span class="o">=</span><span class="s">&#39;\t&#39;</span><span class="p">,</span>header<span class="o">=</span><span class="bp">F</span><span class="p">)[,</span><span class="m">13</span><span class="o">:</span><span class="m">14</span><span class="p">]</span>
<span class="c1"># name the columns</span>
<span class="kp">names</span><span class="p">(</span>annots<span class="p">)</span> <span class="o">&lt;-</span> <span class="kt">c</span><span class="p">(</span><span class="s">&#39;gene&#39;</span><span class="p">,</span><span class="s">&#39;description&#39;</span><span class="p">)</span>
<span class="c1"># combine gene expression and annotations</span>
bothDF_genedesc  <span class="o">&lt;-</span> <span class="kp">merge</span><span class="p">(</span>bothDF<span class="p">,</span>annots<span class="p">,</span> by.x <span class="o">=</span> <span class="s">&quot;Row.names&quot;</span><span class="p">,</span>by.y<span class="o">=</span><span class="s">&#39;gene&#39;</span><span class="p">)</span>
</pre></div>
</div>
<ol class="arabic simple" start="4">
<li>To order your data using FDR, you use the following command in R.</li>
</ol>
<div class="highlight-r"><div class="highlight"><pre><span></span>bothDF_genedesc  <span class="o">&lt;-</span>  bothDF_genedesc<span class="p">[</span><span class="kp">order</span><span class="p">(</span>bothDF_genedesc<span class="o">$</span>FDR<span class="p">),</span> <span class="p">]</span>
</pre></div>
</div>
</div>


           </div>
           <div class="articleComments">
            
           </div>
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
      
        <a href="DE_analysisR.html" class="btn btn-neutral" title="Setup instructions (This is from Data Carpentry (http://www.datacarpentry.org/R-genomics/))" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <p>
        &copy; Copyright 2017, Asela Wijeratne.

    </p>
  </div>
  Built with <a href="http://sphinx-doc.org/">Sphinx</a> using a <a href="https://github.com/snide/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>. 

</footer>

        </div>
      </div>

    </section>

  </div>
  


  

    <script type="text/javascript">
        var DOCUMENTATION_OPTIONS = {
            URL_ROOT:'./',
            VERSION:'170218',
            COLLAPSE_INDEX:false,
            FILE_SUFFIX:'.html',
            HAS_SOURCE:  true,
            SOURCELINK_SUFFIX: '.txt'
        };
    </script>
      <script type="text/javascript" src="_static/jquery.js"></script>
      <script type="text/javascript" src="_static/underscore.js"></script>
      <script type="text/javascript" src="_static/doctools.js"></script>

  

  
  
    <script type="text/javascript" src="_static/js/theme.js"></script>
  

  
  
  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.StickyNav.enable();
      });
  </script>
   

</body>
</html>