

<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <title>BIOINFORMATICS AND APPLICATIONS BIO 4033/ BIO 5033 &mdash; Inclass Activities 170218 documentation</title>
  

  
  
  
  

  

  
  
    

  

  
  
    <link rel="stylesheet" href="_static/css/theme.css" type="text/css" />
  

  

  
        <link rel="index" title="Index"
              href="genindex.html"/>
        <link rel="search" title="Search" href="search.html"/>
    <link rel="top" title="Inclass Activities 170218 documentation" href="#"/>
        <link rel="next" title="1. Multiple Sequence Alignments" href="MEGA_alignment.html"/> 

  
  <script src="_static/js/modernizr.min.js"></script>

</head>

<body class="wy-body-for-nav" role="document">

   
  <div class="wy-grid-for-nav">

    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search">
          

          
            <a href="#" class="icon icon-home"> Inclass Activities
          

          
          </a>

          
            
            
              <div class="version">
                v1
              </div>
            
          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
              
            
            
              <p class="caption"><span class="caption-text">Bioinformatics part I</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="MEGA_alignment.html">1. Multiple Sequence Alignments</a></li>
<li class="toctree-l1"><a class="reference internal" href="phylogenetics.html">2. Steps of building a tree</a></li>
<li class="toctree-l1"><a class="reference internal" href="phylogenetics_2.html">3. Steps of building a tree (Part II)</a></li>
</ul>
<p class="caption"><span class="caption-text">High-throughput data analysis</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="fastQC.html">FastQC analysis using Cyverse Discovery Environment (DE)</a></li>
<li class="toctree-l1"><a class="reference internal" href="relaunch.html">Relaunching a stalled analysis in Cyverse Discovery Environment</a></li>
<li class="toctree-l1"><a class="reference internal" href="adapter_trimming.html">Adapter and quality trimming using trim-galore</a></li>
<li class="toctree-l1"><a class="reference internal" href="mapping.html">Mapping short reads</a></li>
<li class="toctree-l1"><a class="reference internal" href="flag_stat.html">Counting mapped reads</a></li>
<li class="toctree-l1"><a class="reference internal" href="DE_analysis.html">Differential gene expression analysis</a></li>
</ul>
<p class="caption"><span class="caption-text">Protein analysis</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="protein_structure.html">Secondary Structure Prediction</a></li>
<li class="toctree-l1"><a class="reference internal" href="protein_structure.html#tertiary-structure-prediction">Tertiary Structure Prediction</a></li>
</ul>
<p class="caption"><span class="caption-text">qPCR data analysis</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="Delta-Delta_Ct_Method.html">The Delta-Delta Ct Method</a></li>
</ul>
<p class="caption"><span class="caption-text">Galaxy for HTS analysis</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="getting_data.html">Getting data into Galaxy</a></li>
<li class="toctree-l1"><a class="reference internal" href="GalaxyfastQC.html">FastQC analysis using Galaxy</a></li>
<li class="toctree-l1"><a class="reference internal" href="galaxy_adapter_trimming.html">Adapter and quality trimming using Cutadapt</a></li>
<li class="toctree-l1"><a class="reference internal" href="tophat.html">Use Splice aware aligner, Tophat2 to align short reads</a></li>
<li class="toctree-l1"><a class="reference internal" href="htseq.html">Use Htseq to counts reads mapped to features</a></li>
<li class="toctree-l1"><a class="reference internal" href="kellisto.html">Use Kellisto to map reads to cDNA and count</a></li>
<li class="toctree-l1"><a class="reference internal" href="DE_analysisR.html">Setup instructions (This is from Data Carpentry (http://www.datacarpentry.org/R-genomics/))</a></li>
<li class="toctree-l1"><a class="reference internal" href="DE_analysisR.html#using-deseq-and-edger-to-find-differentially-expressed-genes">Using DEseq and EdgeR to find differentially expressed genes</a></li>
<li class="toctree-l1"><a class="reference internal" href="DE_analysisR.html#deseq-analysis">DEseq analysis</a></li>
<li class="toctree-l1"><a class="reference internal" href="DE_analysisR.html#combine-deseq-and-edgr-to-make-venn-diagram">Combine DESeq and EdgR  to make Venn diagram</a></li>
<li class="toctree-l1"><a class="reference internal" href="GO_analysis.html">GOseq analysis</a></li>
</ul>

            
          
        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" role="navigation" aria-label="top navigation">
        
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="#">Inclass Activities</a>
        
      </nav>


      
      <div class="wy-nav-content">
        <div class="rst-content">
          















<div role="navigation" aria-label="breadcrumbs navigation">

  <ul class="wy-breadcrumbs">
    
      <li><a href="#">Docs</a> &raquo;</li>
        
      <li>BIOINFORMATICS AND APPLICATIONS BIO 4033/ BIO 5033</li>
    
    
      <li class="wy-breadcrumbs-aside">
        
            
            <a href="_sources/index.rst.txt" rel="nofollow"> View page source</a>
          
        
      </li>
    
  </ul>

  
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  <div class="section" id="bioinformatics-and-applications-bio-4033-bio-5033">
<h1>BIOINFORMATICS AND APPLICATIONS BIO 4033/ BIO 5033<a class="headerlink" href="#bioinformatics-and-applications-bio-4033-bio-5033" title="Permalink to this headline">¶</a></h1>
<div class="toctree-wrapper compound">
<p class="caption"><span class="caption-text">Bioinformatics part I</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="MEGA_alignment.html">1. Multiple Sequence Alignments</a><ul>
<li class="toctree-l2"><a class="reference internal" href="MEGA_alignment.html#getting-started-with-mega">1.1. Getting started with MEGA</a><ul>
<li class="toctree-l3"><a class="reference internal" href="MEGA_alignment.html#how-to-make-an-alignment-using-mega">1.1.1. How to make an alignment using MEGA</a></li>
<li class="toctree-l3"><a class="reference internal" href="MEGA_alignment.html#edit-the-alignments">1.1.2. Edit the alignments</a></li>
<li class="toctree-l3"><a class="reference internal" href="MEGA_alignment.html#exporting-msa">1.1.3. Exporting MSA</a></li>
</ul>
</li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="phylogenetics.html">2. Steps of building a tree</a><ul>
<li class="toctree-l2"><a class="reference internal" href="phylogenetics.html#make-multiple-sequence-alignment-for-globin-gene-family">2.1. Make multiple sequence alignment for Globin gene family</a></li>
<li class="toctree-l2"><a class="reference internal" href="phylogenetics.html#find-informative-sites-for-parsimony">2.2. Find informative sites for Parsimony</a></li>
<li class="toctree-l2"><a class="reference internal" href="phylogenetics.html#building-phylogenetic-trees">2.3. Building Phylogenetic trees</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="phylogenetics_2.html">3. Steps of building a tree (Part II)</a><ul>
<li class="toctree-l2"><a class="reference internal" href="phylogenetics_2.html#make-multiple-sequence-alignment-for-globin-gene-family">3.1. Make multiple sequence alignment for Globin gene family</a></li>
<li class="toctree-l2"><a class="reference internal" href="phylogenetics_2.html#find-the-best-substitution-model">3.2. Find the best substitution model</a></li>
<li class="toctree-l2"><a class="reference internal" href="phylogenetics_2.html#building-phylogenetic-trees">3.3. Building Phylogenetic trees</a></li>
</ul>
</li>
</ul>
</div>
<div class="toctree-wrapper compound">
<p class="caption"><span class="caption-text">High-throughput data analysis</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="fastQC.html">FastQC analysis using Cyverse Discovery Environment (DE)</a><ul>
<li class="toctree-l2"><a class="reference internal" href="fastQC.html#step-1-login-into-cyverse-de">Step 1: <strong>Login into Cyverse DE</strong></a></li>
<li class="toctree-l2"><a class="reference internal" href="fastQC.html#step-2-getting-data-into-cyverse-discovery-environment">Step 2: <strong>Getting data into Cyverse Discovery Environment</strong></a><ul>
<li class="toctree-l3"><a class="reference internal" href="fastQC.html#urls"><strong>URLs</strong></a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="fastQC.html#step-3-performing-fastqc-analysis">Step 3: <strong>Performing FastQC analysis</strong>:</a></li>
<li class="toctree-l2"><a class="reference internal" href="fastQC.html#reference">Reference:</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="relaunch.html">Relaunching a stalled analysis in Cyverse Discovery Environment</a><ul>
<li class="toctree-l2"><a class="reference internal" href="relaunch.html#step-1-click-on-the-message-icon">Step 1: <strong>Click on the message icon</strong></a></li>
<li class="toctree-l2"><a class="reference internal" href="relaunch.html#step-2-click-on-the-analysis-that-appears-to-be-stalled">Step 2: <strong>Click on the analysis that appears to be stalled</strong></a></li>
<li class="toctree-l2"><a class="reference internal" href="relaunch.html#step-3-check-the-small-box-and-click-on-analysis">Step 3: <strong>Check the small box and click on analysis</strong></a></li>
<li class="toctree-l2"><a class="reference internal" href="relaunch.html#step-4-click-on-the-relaunch-button">Step 4: <strong>Click on the relaunch button</strong></a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="adapter_trimming.html">Adapter and quality trimming using trim-galore</a><ul>
<li class="toctree-l2"><a class="reference internal" href="adapter_trimming.html#step-1-launching-trim-galore">Step 1: <strong>Launching Trim-galore</strong></a></li>
<li class="toctree-l2"><a class="reference internal" href="adapter_trimming.html#step-2-selecting-output-folder">Step 2: <strong>Selecting output folder</strong></a></li>
<li class="toctree-l2"><a class="reference internal" href="adapter_trimming.html#step-3-selecting-input-files">Step 3: <strong>Selecting input files</strong></a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="mapping.html">Mapping short reads</a><ul>
<li class="toctree-l2"><a class="reference internal" href="mapping.html#step-1-mapping-with-tophat2">Step 1: <strong>Mapping with Tophat2</strong></a></li>
<li class="toctree-l2"><a class="reference internal" href="mapping.html#step-2-mapping-with-bowtie2">Step 2: <strong>Mapping with Bowtie2</strong></a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="flag_stat.html">Counting mapped reads</a></li>
<li class="toctree-l1"><a class="reference internal" href="DE_analysis.html">Differential gene expression analysis</a><ul>
<li class="toctree-l2"><a class="reference internal" href="DE_analysis.html#deseq-tutorial">DESeq tutorial:</a></li>
<li class="toctree-l2"><a class="reference internal" href="DE_analysis.html#steps-to-perform-deseq-analysis">Steps to perform DEseq analysis</a></li>
<li class="toctree-l2"><a class="reference internal" href="DE_analysis.html#de-gene-list">DE gene list</a></li>
</ul>
</li>
</ul>
</div>
<div class="toctree-wrapper compound">
<p class="caption"><span class="caption-text">Protein analysis</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="protein_structure.html">Secondary Structure Prediction</a></li>
<li class="toctree-l1"><a class="reference internal" href="protein_structure.html#tertiary-structure-prediction">Tertiary Structure Prediction</a></li>
</ul>
</div>
<div class="toctree-wrapper compound">
<p class="caption"><span class="caption-text">qPCR data analysis</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="Delta-Delta_Ct_Method.html">The Delta-Delta Ct Method</a><ul>
<li class="toctree-l2"><a class="reference internal" href="Delta-Delta_Ct_Method.html#normalization">Normalization</a></li>
<li class="toctree-l2"><a class="reference internal" href="Delta-Delta_Ct_Method.html#average-of-the-control-samples-normal-cells">Average of the control samples (normal cells)</a></li>
<li class="toctree-l2"><a class="reference internal" href="Delta-Delta_Ct_Method.html#calculate-the-ct-relative-to-the-average-of-ct-normal-cells">Calculate the ∆∆Ct relative to the average of ∆Ct normal cells</a></li>
<li class="toctree-l2"><a class="reference internal" href="Delta-Delta_Ct_Method.html#fold-gene-expression-for-each-sample">Fold gene expression for each sample</a></li>
<li class="toctree-l2"><a class="reference internal" href="Delta-Delta_Ct_Method.html#overall-fold-change">Overall fold change</a></li>
<li class="toctree-l2"><a class="reference internal" href="Delta-Delta_Ct_Method.html#log-transformation">Log transformation</a></li>
<li class="toctree-l2"><a class="reference internal" href="Delta-Delta_Ct_Method.html#t-test">T-test</a></li>
</ul>
</li>
</ul>
</div>
<div class="toctree-wrapper compound">
<p class="caption"><span class="caption-text">Galaxy for HTS analysis</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="getting_data.html">Getting data into Galaxy</a><ul>
<li class="toctree-l2"><a class="reference internal" href="getting_data.html#step-1-login-into-galaxy">Step 1: <strong>Login into Galaxy</strong></a></li>
<li class="toctree-l2"><a class="reference internal" href="getting_data.html#step-2-getting-data">Step 2: Getting data</a></li>
<li class="toctree-l2"><a class="reference internal" href="getting_data.html#step-1-click-on-the-upload-icon-on-upper-left-hand-corner">Step 1: Click on the upload icon on upper left hand corner</a></li>
<li class="toctree-l2"><a class="reference internal" href="getting_data.html#step-2-copy-one-of-the-links-above-click-on-the-paste-fetch-icon-and-paste-link-in-the-box-click-on-start">Step 2: Copy one of the links above. Click on the Paste/Fetch icon and paste link in the box. Click on start.</a></li>
<li class="toctree-l2"><a class="reference internal" href="getting_data.html#step-3-one-the-data-is-uploaded-they-will-appear-in-the-right-hand-panel-you-can-use-the-pencil-icon-to-change-the-name">Step 3: One the data is uploaded, they will appear in the right hand panel. You can use the pencil icon to change the name.</a></li>
<li class="toctree-l2"><a class="reference internal" href="getting_data.html#reference">|Reference:</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="GalaxyfastQC.html">FastQC analysis using Galaxy</a><ul>
<li class="toctree-l2"><a class="reference internal" href="GalaxyfastQC.html#step-1-login-into-galaxy">Step 1: <strong>Login into Galaxy</strong></a></li>
<li class="toctree-l2"><a class="reference internal" href="GalaxyfastQC.html#step-3-performing-fastqc-analysis">Step 3: <strong>Performing FastQC analysis</strong>:</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="galaxy_adapter_trimming.html">Adapter and quality trimming using Cutadapt</a><ul>
<li class="toctree-l2"><a class="reference internal" href="galaxy_adapter_trimming.html#step-1-launching-cutadapt-and-performing-the-analysis">Step 1: <strong>Launching Cutadapt and performing the analysis</strong></a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="tophat.html">Use Splice aware aligner, Tophat2 to align short reads</a><ul>
<li class="toctree-l2"><a class="reference internal" href="tophat.html#output-files">Output files:</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="htseq.html">Use Htseq to counts reads mapped to features</a></li>
<li class="toctree-l1"><a class="reference internal" href="kellisto.html">Use Kellisto to map reads to cDNA and count</a></li>
<li class="toctree-l1"><a class="reference internal" href="DE_analysisR.html">Setup instructions (This is from Data Carpentry (http://www.datacarpentry.org/R-genomics/))</a><ul>
<li class="toctree-l2"><a class="reference internal" href="DE_analysisR.html#windows">Windows</a></li>
<li class="toctree-l2"><a class="reference internal" href="DE_analysisR.html#if-you-already-have-r-and-rstudio-installed">If you already have R and RStudio installed</a></li>
<li class="toctree-l2"><a class="reference internal" href="DE_analysisR.html#if-you-dont-have-r-and-rstudio-installed">If you don’t have R and RStudio installed</a></li>
<li class="toctree-l2"><a class="reference internal" href="DE_analysisR.html#macos">macOS</a></li>
<li class="toctree-l2"><a class="reference internal" href="DE_analysisR.html#id1">If you already have R and RStudio installed</a></li>
<li class="toctree-l2"><a class="reference internal" href="DE_analysisR.html#id2">If you don’t have R and RStudio installed</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="DE_analysisR.html#using-deseq-and-edger-to-find-differentially-expressed-genes">Using DEseq and EdgeR to find differentially expressed genes</a></li>
<li class="toctree-l1"><a class="reference internal" href="DE_analysisR.html#deseq-analysis">DEseq analysis</a></li>
<li class="toctree-l1"><a class="reference internal" href="DE_analysisR.html#combine-deseq-and-edgr-to-make-venn-diagram">Combine DESeq and EdgR  to make Venn diagram</a></li>
<li class="toctree-l1"><a class="reference internal" href="GO_analysis.html">GOseq analysis</a></li>
</ul>
</div>
</div>
<div class="section" id="indices-and-tables">
<h1>Indices and tables<a class="headerlink" href="#indices-and-tables" title="Permalink to this headline">¶</a></h1>
<ul class="simple">
<li><a class="reference internal" href="genindex.html"><span class="std std-ref">Index</span></a></li>
<li><a class="reference internal" href="py-modindex.html"><span class="std std-ref">Module Index</span></a></li>
<li><a class="reference internal" href="search.html"><span class="std std-ref">Search Page</span></a></li>
</ul>
</div>


           </div>
           <div class="articleComments">
            
           </div>
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
        <a href="MEGA_alignment.html" class="btn btn-neutral float-right" title="1. Multiple Sequence Alignments" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right"></span></a>
      
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <p>
        &copy; Copyright 2017, Asela Wijeratne.

    </p>
  </div>
  Built with <a href="http://sphinx-doc.org/">Sphinx</a> using a <a href="https://github.com/snide/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>. 

</footer>

        </div>
      </div>

    </section>

  </div>
  


  

    <script type="text/javascript">
        var DOCUMENTATION_OPTIONS = {
            URL_ROOT:'./',
            VERSION:'170218',
            COLLAPSE_INDEX:false,
            FILE_SUFFIX:'.html',
            HAS_SOURCE:  true,
            SOURCELINK_SUFFIX: '.txt'
        };
    </script>
      <script type="text/javascript" src="_static/jquery.js"></script>
      <script type="text/javascript" src="_static/underscore.js"></script>
      <script type="text/javascript" src="_static/doctools.js"></script>

  

  
  
    <script type="text/javascript" src="_static/js/theme.js"></script>
  

  
  
  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.StickyNav.enable();
      });
  </script>
   

</body>
</html>