

<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <title>3. Steps of building a tree (Part II) &mdash; Inclass Activities 170218 documentation</title>
  

  
  
  
  

  

  
  
    

  

  
  
    <link rel="stylesheet" href="_static/css/theme.css" type="text/css" />
  

  

  
        <link rel="index" title="Index"
              href="genindex.html"/>
        <link rel="search" title="Search" href="search.html"/>
    <link rel="top" title="Inclass Activities 170218 documentation" href="index.html"/>
        <link rel="next" title="FastQC analysis using Cyverse Discovery Environment (DE)" href="fastQC.html"/>
        <link rel="prev" title="2. Steps of building a tree" href="phylogenetics.html"/> 

  
  <script src="_static/js/modernizr.min.js"></script>

</head>

<body class="wy-body-for-nav" role="document">

   
  <div class="wy-grid-for-nav">

    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search">
          

          
            <a href="index.html" class="icon icon-home"> Inclass Activities
          

          
          </a>

          
            
            
              <div class="version">
                v1
              </div>
            
          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
              
            
            
              <p class="caption"><span class="caption-text">Bioinformatics part I</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="MEGA_alignment.html">1. Multiple Sequence Alignments</a></li>
<li class="toctree-l1"><a class="reference internal" href="phylogenetics.html">2. Steps of building a tree</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">3. Steps of building a tree (Part II)</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#make-multiple-sequence-alignment-for-globin-gene-family">3.1. Make multiple sequence alignment for Globin gene family</a></li>
<li class="toctree-l2"><a class="reference internal" href="#find-the-best-substitution-model">3.2. Find the best substitution model</a></li>
<li class="toctree-l2"><a class="reference internal" href="#building-phylogenetic-trees">3.3. Building Phylogenetic trees</a></li>
</ul>
</li>
</ul>
<p class="caption"><span class="caption-text">High-throughput data analysis</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="fastQC.html">FastQC analysis using Cyverse Discovery Environment (DE)</a></li>
<li class="toctree-l1"><a class="reference internal" href="relaunch.html">Relaunching a stalled analysis in Cyverse Discovery Environment</a></li>
<li class="toctree-l1"><a class="reference internal" href="adapter_trimming.html">Adapter and quality trimming using trim-galore</a></li>
<li class="toctree-l1"><a class="reference internal" href="mapping.html">Mapping short reads</a></li>
<li class="toctree-l1"><a class="reference internal" href="flag_stat.html">Counting mapped reads</a></li>
<li class="toctree-l1"><a class="reference internal" href="DE_analysis.html">Differential gene expression analysis</a></li>
</ul>
<p class="caption"><span class="caption-text">Protein analysis</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="protein_structure.html">Secondary Structure Prediction</a></li>
<li class="toctree-l1"><a class="reference internal" href="protein_structure.html#tertiary-structure-prediction">Tertiary Structure Prediction</a></li>
</ul>
<p class="caption"><span class="caption-text">qPCR data analysis</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="Delta-Delta_Ct_Method.html">The Delta-Delta Ct Method</a></li>
</ul>
<p class="caption"><span class="caption-text">Galaxy for HTS analysis</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="getting_data.html">Getting data into Galaxy</a></li>
<li class="toctree-l1"><a class="reference internal" href="GalaxyfastQC.html">FastQC analysis using Galaxy</a></li>
<li class="toctree-l1"><a class="reference internal" href="galaxy_adapter_trimming.html">Adapter and quality trimming using Cutadapt</a></li>
<li class="toctree-l1"><a class="reference internal" href="tophat.html">Use Splice aware aligner, Tophat2 to align short reads</a></li>
<li class="toctree-l1"><a class="reference internal" href="htseq.html">Use Htseq to counts reads mapped to features</a></li>
<li class="toctree-l1"><a class="reference internal" href="kellisto.html">Use Kellisto to map reads to cDNA and count</a></li>
<li class="toctree-l1"><a class="reference internal" href="DE_analysisR.html">Setup instructions (This is from Data Carpentry (http://www.datacarpentry.org/R-genomics/))</a></li>
<li class="toctree-l1"><a class="reference internal" href="DE_analysisR.html#using-deseq-and-edger-to-find-differentially-expressed-genes">Using DEseq and EdgeR to find differentially expressed genes</a></li>
<li class="toctree-l1"><a class="reference internal" href="DE_analysisR.html#deseq-analysis">DEseq analysis</a></li>
<li class="toctree-l1"><a class="reference internal" href="DE_analysisR.html#combine-deseq-and-edgr-to-make-venn-diagram">Combine DESeq and EdgR  to make Venn diagram</a></li>
<li class="toctree-l1"><a class="reference internal" href="GO_analysis.html">GOseq analysis</a></li>
</ul>

            
          
        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" role="navigation" aria-label="top navigation">
        
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="index.html">Inclass Activities</a>
        
      </nav>


      
      <div class="wy-nav-content">
        <div class="rst-content">
          















<div role="navigation" aria-label="breadcrumbs navigation">

  <ul class="wy-breadcrumbs">
    
      <li><a href="index.html">Docs</a> &raquo;</li>
        
      <li>3. Steps of building a tree (Part II)</li>
    
    
      <li class="wy-breadcrumbs-aside">
        
            
            <a href="_sources/phylogenetics_2.rst.txt" rel="nofollow"> View page source</a>
          
        
      </li>
    
  </ul>

  
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  <div class="figure align-right">
<img alt="_images/UnivLogo_Stack_2C_Dark.png" src="_images/UnivLogo_Stack_2C_Dark.png" />
</div>
<div class="section" id="steps-of-building-a-tree-part-ii">
<h1>3. Steps of building a tree (Part II)<a class="headerlink" href="#steps-of-building-a-tree-part-ii" title="Permalink to this headline">¶</a></h1>
<div class="section" id="make-multiple-sequence-alignment-for-globin-gene-family">
<h2>3.1. Make multiple sequence alignment for Globin gene family<a class="headerlink" href="#make-multiple-sequence-alignment-for-globin-gene-family" title="Permalink to this headline">¶</a></h2>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Step 1:</th><td class="field-body">Download globin.fasta from Blackboard and perform a MSA using MUSCLE (follow the steps we discussed last week).</td>
</tr>
<tr class="field-even field"><th class="field-name">Step 2:</th><td class="field-body">Examine the alignment to make sure it is correct and no additional editing is needed.</td>
</tr>
<tr class="field-odd field"><th class="field-name">Step 3:</th><td class="field-body">Export the alignment as a fasta format file on your Desktop. Name it as globin_align</td>
</tr>
</tbody>
</table>
<div class="line-block">
<div class="line"><br /></div>
<div class="line"><br /></div>
<div class="line"><br /></div>
</div>
</div>
<div class="section" id="find-the-best-substitution-model">
<h2>3.2. Find the best substitution model<a class="headerlink" href="#find-the-best-substitution-model" title="Permalink to this headline">¶</a></h2>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Step 1:</th><td class="field-body"><p class="first">Calculate the distance using different substitution models
:a: Select <strong>Distance</strong> and then <strong>Compute Pairwise distance</strong>
:b: Calculate distance using the following methods</p>
<blockquote class="last">
<div><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">i:</th><td class="field-body">No. of Differences</td>
</tr>
<tr class="field-even field"><th class="field-name">ii:</th><td class="field-body">p-distances</td>
</tr>
<tr class="field-odd field"><th class="field-name">iii:</th><td class="field-body">Poisson model</td>
</tr>
</tbody>
</table>
</div></blockquote>
</td>
</tr>
</tbody>
</table>
<div class="line-block">
<div class="line"><br /></div>
<div class="line"><br /></div>
</div>
<div class="figure">
<img alt="_images/mega_13.png" src="_images/mega_13.png" />
</div>
<div class="line-block">
<div class="line"><br /></div>
<div class="line"><br /></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Step 2:</th><td class="field-body"><p class="first">Use the same alignment file and build three NJ trees using different substitution models:</p>
<table class="last docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">a:</th><td class="field-body">No. of Differences</td>
</tr>
<tr class="field-even field"><th class="field-name">b:</th><td class="field-body">p-distances</td>
</tr>
<tr class="field-odd field"><th class="field-name">c:</th><td class="field-body">Poisson model</td>
</tr>
</tbody>
</table>
</td>
</tr>
</tbody>
</table>
<div class="line-block">
<div class="line"><br /></div>
<div class="line"><br /></div>
</div>
<div class="figure">
<img alt="_images/mega_12.png" src="_images/mega_12.png" />
</div>
<div class="line-block">
<div class="line"><br /></div>
<div class="line"><br /></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Step 3:</th><td class="field-body"><a class="reference external" href="http://darwin.uvigo.es/cgi-bin/queueSystem.pl?cgi_pid=9928">Best model based on ProtTest</a></td>
</tr>
</tbody>
</table>
</div>
<div class="section" id="building-phylogenetic-trees">
<h2>3.3. Building Phylogenetic trees<a class="headerlink" href="#building-phylogenetic-trees" title="Permalink to this headline">¶</a></h2>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Step 1:</th><td class="field-body">Click on <strong>Phylogeny</strong></td>
</tr>
</tbody>
</table>
<div class="line-block">
<div class="line"><br /></div>
</div>
<div class="figure">
<img alt="_images/mega_10.png" src="_images/mega_10.png" />
</div>
<div class="line-block">
<div class="line"><br /></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Step 2:</th><td class="field-body"><p class="first">Make Neighbor-Joining tree with <strong>Bootstrap 500 replicates</strong></p>
<table class="last docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">A:</th><td class="field-body">What relationships can you see in the tree?</td>
</tr>
<tr class="field-even field"><th class="field-name">B:</th><td class="field-body">What can you say about the statistical support for each relationship?</td>
</tr>
<tr class="field-odd field"><th class="field-name">C:</th><td class="field-body">What should be the out-group?</td>
</tr>
</tbody>
</table>
</td>
</tr>
</tbody>
</table>
<div class="line-block">
<div class="line"><br /></div>
<div class="line"><br /></div>
</div>
<div class="figure">
<img alt="_images/mega_11.png" src="_images/mega_11.png" />
</div>
<div class="line-block">
<div class="line"><br /></div>
<div class="line"><br /></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Step 3:</th><td class="field-body">Save the tree as a pdf file by clicking on <strong>Image</strong> button</td>
</tr>
</tbody>
</table>
<div class="line-block">
<div class="line"><br /></div>
<div class="line"><br /></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Step 4:</th><td class="field-body"><p class="first">Build a tree using Parsimony method with <strong>50 Bootstrap</strong> replicates (500 will be very slow).</p>
<table class="last docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">A:</th><td class="field-body">What relationships can you see in the tree?</td>
</tr>
<tr class="field-even field"><th class="field-name">B:</th><td class="field-body">What can you say about the statistical support for each relationship?</td>
</tr>
<tr class="field-odd field"><th class="field-name">C:</th><td class="field-body">Do you see the same relationships that you saw with NJ tree?</td>
</tr>
</tbody>
</table>
</td>
</tr>
</tbody>
</table>
<div class="line-block">
<div class="line"><br /></div>
<div class="line"><br /></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Step 5:</th><td class="field-body">Bayesian inference of phylogeny</td>
</tr>
</tbody>
</table>
<p><a class="reference external" href="http://www.phylogeny.fr/one_task.cgi?task_type=mrbayes">Follow this link to MrBayes online server</a></p>
<blockquote>
<div><blockquote>
<div><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">A:</th><td class="field-body">Use the same alignment file</td>
</tr>
<tr class="field-even field"><th class="field-name">B:</th><td class="field-body">In MrBayes select Poisson amino acid model with equal rates of substitution.</td>
</tr>
</tbody>
</table>
</div></blockquote>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">C:</th><td class="field-body">Select prior parameters (e.g. equal, fixed frequencies for the states; equal probability for all topologies; unconstrained branch lengths).</td>
</tr>
<tr class="field-even field"><th class="field-name">D:</th><td class="field-body">Run 1,000,000 trials for Monte Carlo Markov Chain estimation of the posterior distribution.</td>
</tr>
<tr class="field-odd field"><th class="field-name">E:</th><td class="field-body">Obtain phylogram</td>
</tr>
<tr class="field-even field"><th class="field-name">F:</th><td class="field-body">Export tree files</td>
</tr>
<tr class="field-odd field"><th class="field-name">G:</th><td class="field-body">View in MEGA software</td>
</tr>
</tbody>
</table>
</div></blockquote>
<div class="line-block">
<div class="line"><br /></div>
<div class="line"><br /></div>
</div>
<div class="figure">
<img alt="_images/mega_15.png" src="_images/mega_15.png" />
</div>
<div class="line-block">
<div class="line"><br /></div>
<div class="line"><br /></div>
</div>
<div class="figure">
<img alt="_images/mega_14.png" src="_images/mega_14.png" />
</div>
</div>
</div>


           </div>
           <div class="articleComments">
            
           </div>
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
        <a href="fastQC.html" class="btn btn-neutral float-right" title="FastQC analysis using Cyverse Discovery Environment (DE)" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right"></span></a>
      
      
        <a href="phylogenetics.html" class="btn btn-neutral" title="2. Steps of building a tree" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <p>
        &copy; Copyright 2017, Asela Wijeratne.

    </p>
  </div>
  Built with <a href="http://sphinx-doc.org/">Sphinx</a> using a <a href="https://github.com/snide/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>. 

</footer>

        </div>
      </div>

    </section>

  </div>
  


  

    <script type="text/javascript">
        var DOCUMENTATION_OPTIONS = {
            URL_ROOT:'./',
            VERSION:'170218',
            COLLAPSE_INDEX:false,
            FILE_SUFFIX:'.html',
            HAS_SOURCE:  true,
            SOURCELINK_SUFFIX: '.txt'
        };
    </script>
      <script type="text/javascript" src="_static/jquery.js"></script>
      <script type="text/javascript" src="_static/underscore.js"></script>
      <script type="text/javascript" src="_static/doctools.js"></script>

  

  
  
    <script type="text/javascript" src="_static/js/theme.js"></script>
  

  
  
  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.StickyNav.enable();
      });
  </script>
   

</body>
</html>