

<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <title>Use Splice aware aligner, Tophat2 to align short reads &mdash; Inclass Activities 170218 documentation</title>
  

  
  
  
  

  

  
  
    

  

  
  
    <link rel="stylesheet" href="_static/css/theme.css" type="text/css" />
  

  

  
        <link rel="index" title="Index"
              href="genindex.html"/>
        <link rel="search" title="Search" href="search.html"/>
    <link rel="top" title="Inclass Activities 170218 documentation" href="index.html"/>
        <link rel="next" title="Use Htseq to counts reads mapped to features" href="htseq.html"/>
        <link rel="prev" title="Adapter and quality trimming using Cutadapt" href="galaxy_adapter_trimming.html"/> 

  
  <script src="_static/js/modernizr.min.js"></script>

</head>

<body class="wy-body-for-nav" role="document">

   
  <div class="wy-grid-for-nav">

    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search">
          

          
            <a href="index.html" class="icon icon-home"> Inclass Activities
          

          
          </a>

          
            
            
              <div class="version">
                v1
              </div>
            
          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
              
            
            
              <p class="caption"><span class="caption-text">Bioinformatics part I</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="MEGA_alignment.html">1. Multiple Sequence Alignments</a></li>
<li class="toctree-l1"><a class="reference internal" href="phylogenetics.html">2. Steps of building a tree</a></li>
<li class="toctree-l1"><a class="reference internal" href="phylogenetics_2.html">3. Steps of building a tree (Part II)</a></li>
</ul>
<p class="caption"><span class="caption-text">High-throughput data analysis</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="fastQC.html">FastQC analysis using Cyverse Discovery Environment (DE)</a></li>
<li class="toctree-l1"><a class="reference internal" href="relaunch.html">Relaunching a stalled analysis in Cyverse Discovery Environment</a></li>
<li class="toctree-l1"><a class="reference internal" href="adapter_trimming.html">Adapter and quality trimming using trim-galore</a></li>
<li class="toctree-l1"><a class="reference internal" href="mapping.html">Mapping short reads</a></li>
<li class="toctree-l1"><a class="reference internal" href="flag_stat.html">Counting mapped reads</a></li>
<li class="toctree-l1"><a class="reference internal" href="DE_analysis.html">Differential gene expression analysis</a></li>
</ul>
<p class="caption"><span class="caption-text">Protein analysis</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="protein_structure.html">Secondary Structure Prediction</a></li>
<li class="toctree-l1"><a class="reference internal" href="protein_structure.html#tertiary-structure-prediction">Tertiary Structure Prediction</a></li>
</ul>
<p class="caption"><span class="caption-text">qPCR data analysis</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="Delta-Delta_Ct_Method.html">The Delta-Delta Ct Method</a></li>
</ul>
<p class="caption"><span class="caption-text">Galaxy for HTS analysis</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="getting_data.html">Getting data into Galaxy</a></li>
<li class="toctree-l1"><a class="reference internal" href="GalaxyfastQC.html">FastQC analysis using Galaxy</a></li>
<li class="toctree-l1"><a class="reference internal" href="galaxy_adapter_trimming.html">Adapter and quality trimming using Cutadapt</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">Use Splice aware aligner, Tophat2 to align short reads</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#output-files">Output files:</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="htseq.html">Use Htseq to counts reads mapped to features</a></li>
<li class="toctree-l1"><a class="reference internal" href="kellisto.html">Use Kellisto to map reads to cDNA and count</a></li>
<li class="toctree-l1"><a class="reference internal" href="DE_analysisR.html">Setup instructions (This is from Data Carpentry (http://www.datacarpentry.org/R-genomics/))</a></li>
<li class="toctree-l1"><a class="reference internal" href="DE_analysisR.html#using-deseq-and-edger-to-find-differentially-expressed-genes">Using DEseq and EdgeR to find differentially expressed genes</a></li>
<li class="toctree-l1"><a class="reference internal" href="DE_analysisR.html#deseq-analysis">DEseq analysis</a></li>
<li class="toctree-l1"><a class="reference internal" href="DE_analysisR.html#combine-deseq-and-edgr-to-make-venn-diagram">Combine DESeq and EdgR  to make Venn diagram</a></li>
<li class="toctree-l1"><a class="reference internal" href="GO_analysis.html">GOseq analysis</a></li>
<li class="toctree-l1"><a class="reference internal" href="workflow.html">Run RNAseq analysis as a workflow</a></li>
</ul>

            
          
        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" role="navigation" aria-label="top navigation">
        
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="index.html">Inclass Activities</a>
        
      </nav>


      
      <div class="wy-nav-content">
        <div class="rst-content">
          















<div role="navigation" aria-label="breadcrumbs navigation">

  <ul class="wy-breadcrumbs">
    
      <li><a href="index.html">Docs</a> &raquo;</li>
        
      <li>Use Splice aware aligner, Tophat2 to align short reads</li>
    
    
      <li class="wy-breadcrumbs-aside">
        
            
            <a href="_sources/tophat.rst.txt" rel="nofollow"> View page source</a>
          
        
      </li>
    
  </ul>

  
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  <div class="figure align-right">
<img alt="_images/UnivLogo_Stack_2C_Dark.png" src="_images/UnivLogo_Stack_2C_Dark.png" />
</div>
<div class="section" id="use-splice-aware-aligner-tophat2-to-align-short-reads">
<h1>Use Splice aware aligner, Tophat2 to align short reads<a class="headerlink" href="#use-splice-aware-aligner-tophat2-to-align-short-reads" title="Permalink to this headline">¶</a></h1>
<ol class="arabic">
<li><p class="first">We are using Paired end reads and set the &#8220;Is this library mate-paired?&#8221; pulldown to &#8220;Paired-end&#8221;, then a second pulldown will appear to specify the 2nd FASTQ.</p>
</li>
<li><p class="first">“Mean Inner Distance between Mate Pairs”  value for this parameter should obtained from the person incharge of the sequencing.</p>
</li>
<li><p class="first">Mean Inner Distance between Mate Pairs = length of the Fragments used for sequencing – (Length of Illumina adapters (often 120bp) + part sequenced (76+76))</p>
</li>
<li><p class="first">Genome should be obtained from the SolGenome.net (<a class="reference external" href="ftp://ftp.solgenomics.net/tomato_genome/assembly/build_3.00/">ftp://ftp.solgenomics.net/tomato_genome/assembly/build_3.00/</a>) and select it from the history.</p>
<blockquote>
<div><div class="figure">
<img alt="_images/galaxy_tophat_1.png" src="_images/galaxy_tophat_1.png" />
</div>
</div></blockquote>
</li>
<li><p class="first">This library has been prepared to preserve the strandedness of the RNAs.</p>
<blockquote>
<div><div class="figure">
<img alt="_images/galaxy_tophat_2.png" src="_images/galaxy_tophat_2.png" />
</div>
</div></blockquote>
</li>
</ol>
<div class="line-block">
<div class="line"><br /></div>
</div>
<ol class="arabic" start="6">
<li><p class="first">Minimum and maximum intron lengths should be changed according to genome used.</p>
<div class="figure">
<img alt="_images/galaxy_tophat_3.png" src="_images/galaxy_tophat_3.png" />
</div>
</li>
</ol>
<div class="line-block">
<div class="line"><br /></div>
</div>
<ol class="arabic" start="7">
<li><p class="first">Change the intron lengths for split reads as well.</p>
<div class="figure">
<img alt="_images/galaxy_tophat_4.png" src="_images/galaxy_tophat_4.png" />
</div>
</li>
</ol>
<div class="line-block">
<div class="line"><br /></div>
</div>
<div class="section" id="output-files">
<h2>Output files:<a class="headerlink" href="#output-files" title="Permalink to this headline">¶</a></h2>
<ol class="arabic simple">
<li>accepted_hits (BAM, BAI)</li>
</ol>
<div class="line-block">
<div class="line"><br /></div>
</div>
<ol class="arabic simple" start="2">
<li>Two binary files: .BAM (data) and .BAI (index)</li>
</ol>
<div class="line-block">
<div class="line"><br /></div>
</div>
<p>3. These are the actual paired reads mapped to their position on the genome, and split across exon junctions. This can be visualized in IGV, IGB or UCSC, but you must download both .BAM and .BAI files to the same directory.
splice_junctions (BED)</p>
<div class="line-block">
<div class="line"><br /></div>
</div>
<ol class="arabic simple" start="4">
<li>BED file (list of genomic locations, no sequence) listing all the places TopHat had to split a read into two pieces to span an exon junction. This can be visualized at UCSC or in IGV, etc.</li>
</ol>
<div class="line-block">
<div class="line"><br /></div>
</div>
<ol class="arabic simple" start="5">
<li>deletions (BED) (if indel search is on)</li>
</ol>
<div class="line-block">
<div class="line"><br /></div>
</div>
<ol class="arabic simple" start="6">
<li>insertions (BED) (if indel search is on)</li>
</ol>
</div>
</div>


           </div>
           <div class="articleComments">
            
           </div>
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
        <a href="htseq.html" class="btn btn-neutral float-right" title="Use Htseq to counts reads mapped to features" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right"></span></a>
      
      
        <a href="galaxy_adapter_trimming.html" class="btn btn-neutral" title="Adapter and quality trimming using Cutadapt" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <p>
        &copy; Copyright 2017, Asela Wijeratne.

    </p>
  </div>
  Built with <a href="http://sphinx-doc.org/">Sphinx</a> using a <a href="https://github.com/snide/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>. 

</footer>

        </div>
      </div>

    </section>

  </div>
  


  

    <script type="text/javascript">
        var DOCUMENTATION_OPTIONS = {
            URL_ROOT:'./',
            VERSION:'170218',
            COLLAPSE_INDEX:false,
            FILE_SUFFIX:'.html',
            HAS_SOURCE:  true,
            SOURCELINK_SUFFIX: '.txt'
        };
    </script>
      <script type="text/javascript" src="_static/jquery.js"></script>
      <script type="text/javascript" src="_static/underscore.js"></script>
      <script type="text/javascript" src="_static/doctools.js"></script>

  

  
  
    <script type="text/javascript" src="_static/js/theme.js"></script>
  

  
  
  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.StickyNav.enable();
      });
  </script>
   

</body>
</html>